Modules/SubModule_ASL/xASL_wrp_ResampleASL.m

```diff
@@ -244,15 +244,23 @@ function xASL_wrp_ResampleASL(x)
     elseif x.modules.asl.bTimeEncoded
         % Decoding of TimeEncoded data (Nifti is saved inside the function)
         ASL_im = xASL_quant_HadamardDecoding(PathASL{iSpace}, x.Q);
+        PLD_vector_size = length(unique(x.Q.Initial_PLD));
+        blockSize = x.Q.NumberEchoTimes * (PLD_vector_size - (PLD_vector_size/x.Q.TimeEncodedMatrixSize)); % Number of volumes per Hadamard block = EchoTimes*DecodedMatrixSize

```

Comment:
Please add (or correct first):
"A Hadamard block is a individual labeling duration unit in the Hadamard labeling matrix design."

Reply:
Fixed in 9f3ee421
Explained in a comment below.